assets/js/modules/adsense/components/module/ModuleOverviewWidget/Overview.js

```diff
@@ -46,75 +47,43 @@ function Overview( {
-						) }
-						changeDataUnit="%"
-						context="button"
-						selected={ selectedStats === 0 }
-						handleStatSelection={ handleStatsSelection }
-					/>
-				</Cell>
-				<Cell { ...cellProps }>
-					<DataBlock
-						stat={ 1 }
-						className="googlesitekit-data-block--page-rpm googlesitekit-data-block--button-2"
-						title={ metrics[ headers[ 1 ].name ] }
-						datapoint={ totals?.cells[ 1 ].value || 0 }
-						datapointUnit={ headers[ 1 ]?.currencyCode }
-						change={ calculateChange(
-							previousTotals?.cells[ 1 ].value || 0,
-							totals?.cells[ 1 ].value || 0
-						) }
-						changeDataUnit="%"
-						context="button"
-						selected={ selectedStats === 1 }
-						handleStatSelection={ handleStatsSelection }
-					/>
-				</Cell>
-				<Cell { ...cellProps }>
-					<DataBlock
-						stat={ 2 }
-						className="googlesitekit-data-block--page-rpm googlesitekit-data-block--button-3"
-						title={ metrics[ headers[ 2 ].name ] }
-						datapoint={ totals?.cells[ 2 ].value || 0 }
-						change={ calculateChange(
-							previousTotals?.cells[ 2 ].value || 0,
-							totals?.cells[ 2 ].value || 0
-						) }
-						changeDataUnit="%"
-						context="button"
-						selected={ selectedStats === 2 }
-						handleStatSelection={ handleStatsSelection }
-					/>
-				</Cell>
-				<Cell { ...cellProps }>
-					<DataBlock
-						stat={ 3 }
-						className="googlesitekit-data-block--impression googlesitekit-data-block--button-4"
-						title={ metrics[ headers[ 3 ].name ] }
-						datapoint={ totals?.cells[ 3 ].value || 0 }
-						datapointUnit="%"
-						change={ calculateChange(
-							previousTotals?.cells[ 3 ].value || 0,
-							totals?.cells[ 3 ].value || 0
-						) }
-						changeDataUnit="%"
-						context="button"
-						selected={ selectedStats === 3 }
-						handleStatSelection={ handleStatsSelection }
-					/>
-				</Cell>
-			</Row>
+			<DataBlockGroup className="mdc-layout-grid__inner">
+				{ headers?.map( ( headerData, index ) => {
+					let datapointUnit = null;
+					if ( index === 0 || index === 1 ) {
+						datapointUnit = headerData?.currencyCode;
+					}
+					if ( index === 3 ) {
+						datapointUnit = '%';
+					}
+					return (
+						<Cell
+							key={ metrics[ headers[ index ].name ] }
+							{ ...cellProps }
+						>
+							<DataBlock
+								stat={ index }
+								className={ `googlesitekit-data-block--${
+									index === 3 ? 'impression' : 'page-rpm'
+								} googlesitekit-data-block--button-${
+									index + 1
+								}` }
+								title={ metrics[ headerData.name ] }
+								datapoint={ totals?.cells[ index ].value || 0 }
+								datapointUnit={ datapointUnit }
+								change={ calculateChange(
+									previousTotals?.cells[ index ].value || 0,
+									totals?.cells[ index ].value || 0
+								) }
+								changeDataUnit="%"
+								context="button"
+								selected={ selectedStats === index }
+								handleStatSelection={ handleStatsSelection }
+							/>
+						</Cell>
```

Comment:
Why this change? It's less code, but it is much less readable as 4 different datapoints, and the `index === 3 ?` conditions seem unmaintainable; if we end up with a new type of value in here the conditionals will get more verbose and harder to read.

Let's restore this to four separate `DataBlock`s with their own distinct values.

If these could be looped through without any conditionals I'd be fine with this change, but this ends up being harder to parse and less flexible to maintain.

Reply:
Got it, that's a good point. Restored